### ISSUE TITLE
Redirect behaviour change on authorize in 5.6.0

### ISSUE THREAD

**Issue by j-seixas** [doorkeeper-gem/doorkeeper#1616]
### Steps to reproduce
- Config with `use_refresh_token`
- Have an authorized application with expired token
- GET `authorize`

### Expected behavior
When performing `authorize` request if a previous token already existed (`matching_token?` would return true even with expired tokens and it makes sense as authorization was given regardless of expiration date of the last token) it redirected to the application redirect_uri. 

### Actual behavior
Since #1542, it only searches for active tokens, which doesn't make sense specially when using use_refresh_token config, as it is now redirecting and requesting a new authorization as it doesn't find any "active" tokens.

### System configuration

**Doorkeeper initializer**:
```ruby
# config/initializers/doorkeeper.rb
Doorkeeper.configure do
  orm :active_record
  resource_owner_authenticator do
    ...
  end
  use_polymorphic_resource_owner
  use_refresh_token
  optional_scopes :scope1, :scope2 ...
  # no default scopes
end
```

**Ruby version**: `3.2.0` (and `3.1.0`)
**Doorkeeper version**: `5.6.0` and later

**Comment by nbulaj** [comment]
Thanks @j-seixas for reporting this. Have to review the original MR. I remember a lot applications faced troubles with CPU / memory usage when deal with a ton of access tokens, that's why improvements around it were made. But they could ignore the case you've described so additional checks should be added. I'll revisit them 

You're welcome with the PR if you have more vision / ideas for this issues :bow:

**Comment by nbulaj** [comment]
Looks like not all the places should check for non expired access token, tried to fix it in https://github.com/doorkeeper-gem/doorkeeper/pull/1631

Could you please check from this branch when you have time? Thanks!

**Comment by j-seixas** [comment]
@nbulaj Seems to work properly on that branch! Thanks!

**Comment by nbulaj** [comment]
Thanks @j-seixas :bow:

**Comment by nbulaj** [comment]
Released as 5.6.4
